### PR TITLE
[#27] unknown providers and relation targets are added on the fly

### PR DESCRIPTION
**Pull request by bonndan**

### CODE REVIEW

File: src/main/java/de/bonndan/nivio/input/InstantItemResolver.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    public InstantItemResolver(ProcessLog log) {
+        this.log = log;
+    }
+
+    public void processTargets(LandscapeDescription landscape) {
+        List<ItemDescription> all = landscape.getItemDescriptions();
+
+        if (!landscape.getConfig().isGreedy())
+            return;
```
Comment (basseur) [inline src/main/java/de/bonndan/nivio/input/InstantItemResolver.java]: This should be the first thing to be called before instantiating `List<ItemDescription> all`.

Also, I personally would never omit curly braces. For guard clauses like this, it _may_ be okay, if you move the return onto the same line.

Comment (bonndan) [reply src/main/java/de/bonndan/nivio/input/InstantItemResolver.java]: :+1:

File: src/main/java/de/bonndan/nivio/input/Indexer.java [hunk 2]
```diff
@@ -53,6 +53,7 @@ public ProcessLog reIndex(final LandscapeDescription input) {
             Map<ItemDescription, List<String>> templatesAndTargets = new HashMap<>();
             new SourceReferencesResolver(formatFactory, logger).resolve(input, templatesAndTargets);
             new TemplateResolver().processTemplates(input, templatesAndTargets);
+            new InstantItemResolver(logger).processTargets(input);
```
Comment (bonndan) [inline src/main/java/de/bonndan/nivio/input/Indexer.java]: @basseur @Matthimatiker @sl0815 

Any suitable patterns here?

Comment (basseur) [reply src/main/java/de/bonndan/nivio/input/Indexer.java]: What do you mean? 😬

Comment (bonndan) [reply src/main/java/de/bonndan/nivio/input/Indexer.java]: new X, new Y, new Z ... and probably the group creation below might be handled by a class, too.

**Pull request merged**